nixos/tests/systemd.nix

```diff
@@ -49,6 +49,25 @@ import ./make-test-python.nix ({ pkgs, ... }: {
       '';
     };
 
+    systemd.services.test-options-merging = {
+      description = "Test Options Merging";
+      wantedBy = [ "multi-user.target" ];
+      script = ''
+        set -eux
+        test -d /var/lib/test-options-merging/foo
+        test -d /var/lib/test-options-merging/bar
+      '';
+      serviceConfig = lib.mkMerge [
+        {
+          Type = "oneshot";
+          RemainAfterExit = true;
+        }
+        # Merging must lift non-list values to a list if at least of them is a list.
```

Comment:
I'd probably also test whether eg. having two non-list definitions will still result in an error about a conflicting definition, because that's something we certainly want to avoid here. Having such a test would involve `builtins.tryEval` which I think shouldn't belong to a VM test, so I'd probably put it in something like `pkgs/test/systemd.nix`.

Reply:
Right, I've added a test for a merge conflict. In the same file `nixos/tests/systemd.nix` though.